yt/yt/tests/integration/queues/test_queue_agent.py

```diff
@@ -3235,6 +3146,142 @@ def test_multicell_export(self, queue_external_cell_tag):
 
     # TODO(achulkov2): Add test that replicated/chaos queues are not exported.
 
+    @authors("apachee")
+    def test_long_exports(self):
+        # Just a little sanity check to at least somewhat verify that we do not export data that is not yet ready, e.g.
+        # we do not make daily export for this day before the end of it.
+        # This test assumes making an export does not take too long (less than a couple of seconds).
+
+        _, queue_id = self._create_queue("//tmp/q")
+
+        export_dir = "//tmp/export"
+        self._create_export_destination(export_dir, queue_id)
+
+        export_period_seconds = 15
+
+        set("//tmp/q/@static_export_config", {
+            "default": {
+                "export_directory": export_dir,
+                "export_period": export_period_seconds * 1000,
+            }
+        })
+
+        orchid = QueueAgentOrchid()
+        self._wait_for_component_passes()
+        orchid.get_queue_orchid("primary://tmp/q").wait_fresh_pass()
+
+        self._sleep_until_next_export_instant(export_period_seconds, 1)
+        offseted_iteration_start = time.time()
+
+        insert_rows("//tmp/q", [{"data": "vim"}] * 2)
+        self._flush_table("//tmp/q")
+
+        insert_rows("//tmp/q", [{"data": "notepad"}] * 2)
+        self._flush_table("//tmp/q")
+
+        iterations = 0
+        assert export_period_seconds >= 10
+        while time.time() - offseted_iteration_start < export_period_seconds - 5:
+            iterations += 1
+            assert len(ls(export_dir)) == 0
+            time.sleep(0.5)
+        assert iterations > 0
+
+        self._sleep_until_next_export_instant(export_period_seconds, 1)
+        wait(lambda: len(ls(export_dir)) == 1)
+
+        self.remove_export_destination(export_dir)
+
+    @authors("apachee")
+    def test_export_retries(self):
+        _, queue_id = self._create_queue("//tmp/q")
+
+        export_dir = "//tmp/export"
+        self._create_export_destination(export_dir, queue_id)
+
+        export_period_seconds = 5
+        set("//tmp/q/@static_export_config", {
+            "default": {
+                "export_directory": export_dir,
+                "export_period": export_period_seconds * 1000,
+            }
+        })
+
+        tx = start_transaction()
+        lock(export_dir, mode="shared", tx=tx, attribute_key="queue_static_exporter")
+
+        insert_rows("//tmp/q", [{"data": "sample"}] * 2)
+        self._flush_table("//tmp/q")
+
+        self._sleep_until_next_export_instant(export_period_seconds)
+        time.sleep(export_period_seconds)
+
+        # XXX(apachee): Is it really enough to check that retry is scheduled way before the next export iteration?
```

Comment:
let's add a test where we wait longer, write/flush some more data and then see how it has to retry creating the first table and also create a new one